nixnet/types.py

```diff
@@ -220,3 +221,110 @@ def __repr__(self):
             self.echo,
             self.type,
             self.timestamp)
+
+
+class LinFrame(object):
+    """LIN Frame.
+
+    Attributes:
+        identifier: An integer representing the LIN frame arbitration identifier.
+        echo: A boolean indicating if the frame is an echo of a successful
+            transmit rather than being received from the network.
+        type: Frame type as described in :any:`nixnet._enums.FrameType`.
+        timestamp: Absolute time the XNET interface received the end-of-frame.
+        payload: A byte string representing the payload.
+        eventid: Identifier for an event-triggered slot.
+        eventslot: A boolean indicating whether the frame was received within an
+            event-triggered slot or an unconditional or sporadic slot.
+    """
+
+    __slots__ = [
+        "identifier",
+        "echo",
+        "type",
+        "timestamp",
+        "eventid",
+        "eventslot",
+        "payload"]
+
+    _FRAME_ID_MASK = 0x000000FF
+
+    def __init__(self, identifier, type, eventid=b"", payload=b""):
+        # type: (int, constants.FrameType, bytes, bytes) -> None
+        self.identifier = identifier
+        self.echo = False  # Used only for Read
+        self.type = type
+        self.timestamp = 0  # Used only for Read
+        self.eventid = eventid  # Used only for Read
+        self.eventslot = False  # Used only for Read
```

Comment:
`eventid` is allowed to be passed in but not `eventslot`?  If `eventid` has anything, then `eventslot` should be `True`.

Maybe not include `eventid` in the parameter list and hard code it to `None` so that it works well for people creating a `LinFrame` for writing (where they'll want a payload)